app/src/androidTest/java/ch/epfl/sdp/lobby/game/MockGameLobbyRepositoryTest.kt

```diff
@@ -0,0 +1,101 @@
+package ch.epfl.sdp.lobby.game
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import ch.epfl.sdp.game.data.Participation
+import ch.epfl.sdp.lobby.PlayerParametersFragment
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+
+@RunWith(AndroidJUnit4::class)
+class MockGameLobbyRepositoryTest {
+    private val glr = MockGameLobbyRepository
+
+    @Test
+    fun getGameIdCallbackIsCalled() {
+        var called = false
+        glr.getGameId { called = true }
+        assertTrue(called)
+    }
+
+    @Test
+    fun getGameNameCallbackIsCalled() {
+        var called = false
+        glr.getGameName { called = true }
+        assertTrue(called)
+    }
+
+    @Test
+    fun getGameDurationCallbackIsCalled() {
+        var called = false
+        glr.getGameDuration { called = true }
+        assertTrue(called)
+    }
+
+    @Test
+    fun getPlayersCallbackIsCalled() {
+        var called = false
+        glr.getPlayers { called = true }
+        assertTrue(called)
+    }
+
+    @Test
+    fun getAdminIdCallbackIsCalled() {
+        var called = false
+        glr.getAdminId { called = true }
+        assertTrue(called)
+    }
+
+    @Test
+    fun getPlayersAddsAPlayer() {
+        var nPlayers = 0
+        glr.getPlayers { players -> nPlayers = players.size }
+        var newNPlayers = 0
+        glr.getPlayers { players -> newNPlayers = players.size }
+        assertEquals(nPlayers + 1, newNPlayers)
+    }
+
+    @Test
+    fun changePlayerReadyChangesReady() {
+        var playerList = Collections.emptyList<Participation>()
+        glr.getPlayers {
+            players -> playerList = players
+        }
+        val player1IsReady = playerList[3].ready
+        val player2IsReady = playerList[4].ready
+        glr.changePlayerReady(playerList[3].user.uid)
+        glr.changePlayerReady(playerList[4].user.uid)
+
+        glr.getPlayers {
+            players -> playerList = players
+        }
+
+        val newPlayer1IsReady = playerList[3].ready
+        val newPlayer2IsReady = playerList[4].ready
+
+
+        assertEquals(player1IsReady, !newPlayer1IsReady)
+        assertEquals(player2IsReady, !newPlayer2IsReady)
```

Comment:
Better change one player from ready to not ready and the other one from not ready to ready, so you check both comportments

Reply:
It is what is done here actually. The method changePlayerReady() just inverts the ready attribute